tensornetwork/ncon_interface_test.py

```diff
@@ -23,121 +23,267 @@
 from tensornetwork.contractors.naive_contractor import naive
 
 
-def test_sanity_check():
+def test_sanity_check_numpy_backend():
+  result = ncon_interface.ncon([np.ones(
+      (2, 2)), np.ones((2, 2))], [(-1, 1), (1, -2)], backend='numpy')
+  np.testing.assert_allclose(result, np.ones((2, 2)) * 2)
+
+
+def test_order_spec_numpy_backend():
```

Comment:
Add `backend` as an argument to each of these tests. It'll automatically test all 4 of our main backends.

Reply:
Did this. One test fails for Jax. I can somehow see why, but don't understand why it does not happen for NumPy. It goes deeper into the contraction methods. I can't quick fix it, so I think it is better to make a separate issue.

Reply:
We should really fix it then, as a failure of JAX with ncon is a big dea.

Reply:
Actually, I changed my mind and I see your point.

Martin has some fixes for `ncon_interface_test` in his PR https://github.com/google/TensorNetwork/pull/264. Would you be ok if you revert the `ncon_interface_test` file for this PR? I really like your other fixes and I think we should have them, but ncon is a little messier than anyone expected.

Reply:
I'll revert this, let's use Martin's fix!

Reply:
Thank you!